src/Protocol/ActivityPub/Queue.php

```diff
@@ -224,6 +226,13 @@ public static function clear()
 		// We delete all entries that aren't associated with a worker entry after seven days.
 		// The other entries are deleted when the worker deferred for too long.
 		DBA::delete('inbox-entry', ["`wid` IS NULL AND `received` < ?", DateTimeFormat::utc('now - 7 days')]);
+
```

Comment:
Shouldn't this be done by a cron-job rather than "on-the-fly"? Optimizing tables cost some time and CPU cycles on PHP and database end.

Reply:
The function `Queue::clear()` is called by the workerqueue task "Cron" and is executed hourly in the background.

Reply:
Okay, thank you for explaining it. :-)